SetupConfigure.cmake

```diff
@@ -0,0 +1,236 @@
+endif()
+
+###########################################
+# Setup paths
+###########################################
+if (OS_IS_MAC)
+    SET(Mscore_INSTALL_NAME  "Contents/Resources/")
+    SET(Mscore_SHARE_NAME    "mscore.app/")
+elseif (OS_IS_WIN)
+    SET(Mscore_INSTALL_NAME  "")
+    SET(Mscore_SHARE_NAME    "./")
+else()
+    SET(Mscore_INSTALL_NAME  "mscore${MUSESCORE_INSTALL_SUFFIX}-${MUSESCORE_VERSION}/")
+    SET(Mscore_SHARE_NAME    "share/")
+endif()
+
+###########################################
+# CONFIGURE: Desktop App
+###########################################
+set(MUE_GENERAL_APP OFF)
+if(BUILD_CONFIGURE MATCHES "APP")
+    set(MUE_GENERAL_APP ON)
+endif()
+
+if(BUILD_CONFIGURE MATCHES "APP-PORTABLE")
+    set(MUE_GENERAL_APP ON)
+    set(WIN_PORTABLE ON)
+endif()
+
+if (MUE_GENERAL_APP)
+    if (BUILD_IS_DEBUG)
+        set(MUE_ENABLE_LOGGER_DEBUGLEVEL ON)
+    else()
+        set(MUE_ENABLE_LOGGER_DEBUGLEVEL OFF)
+    endif()
+endif()
+
+if (OS_IS_MAC OR OS_IS_WIN)
+    if (WIN_PORTABLE)
+        set(MUE_BUILD_UPDATE_MODULE OFF)
+    endif()
+endif()
+
+###########################################
+# CONFIGURE: VTest
+###########################################
+if(BUILD_CONFIGURE MATCHES "VTEST")
+    set(MUE_BUILD_AUDIO_MODULE OFF)
+    set(MUE_BUILD_MUSESAMPLER_MODULE OFF)
+    set(MUE_BUILD_NETWORK_MODULE OFF)
+    set(MUE_BUILD_SHORTCUTS_MODULE OFF)
+    set(MUE_BUILD_VST_MODULE OFF)
+    set(MUE_BUILD_CLOUD_MODULE OFF)
+    set(MUE_BUILD_IMPORTEXPORT_MODULE OFF)
+    set(MUE_BUILD_VIDEOEXPORT_MODULE OFF)
+    set(MUE_BUILD_INSPECTOR_MODULE OFF)
+    set(MUE_BUILD_INSTRUMENTSSCENE_MODULE OFF)
+    set(MUE_BUILD_LANGUAGES_MODULE OFF)
+    set(MUE_BUILD_LEARN_MODULE OFF)
+    set(MUE_BUILD_MULTIINSTANCES_MODULE OFF)
+    set(MUE_BUILD_PALETTE_MODULE OFF)
+    set(MUE_BUILD_PLAYBACK_MODULE OFF)
+    set(MUE_BUILD_PLUGINS_MODULE OFF)
+    set(MUE_BUILD_UPDATE_MODULE OFF)
+    set(MUE_BUILD_WORKSPACE_MODULE OFF)
+
+    set(MUE_BUILD_UNIT_TESTS OFF)
+    set(MUE_ENABLE_LOGGER_DEBUGLEVEL ON)
+endif()
+
+###########################################
+# CONFIGURE: UTest
+###########################################
+if(BUILD_CONFIGURE MATCHES "UTEST")
+    set(MUE_BUILD_UNIT_TESTS ON)
+    set(MUE_ENABLE_LOGGER_DEBUGLEVEL ON)
+endif()
+
+###########################################
+# Subsystem
+###########################################
+
+set(QT_SUPPORT ON)
+
+if (NOT MUE_BUILD_AUDIO_MODULE)
+    set(MUE_BUILD_MUSESAMPLER_MODULE OFF)
+    set(MUE_BUILD_VST_MODULE OFF)
+endif()
+
+if (NOT MUE_BUILD_IMPORTEXPORT_MODULE)
+    set(MUE_BUILD_VIDEOEXPORT_MODULE OFF)
+endif()
+
+if (NOT MUE_BUILD_DIAGNOSTICS_MODULE)
+    set(MUE_BUILD_CRASHPAD_CLIENT OFF)
+endif()
+
+if (MUE_BUILD_ASAN)
+    set(MUE_ENABLE_CUSTOM_ALLOCATOR OFF)
+endif()
+
+###########################################
+# Global definitions
+###########################################
+
+add_definitions(-DMUSESCORE_REVISION="${MUSESCORE_REVISION}")
```

Comment:
I chose the decision not in favor of `config.h` for reasons: 
1. I would like to explicitly separate global defines and local ones, for a specific module, and minimize the number of global ones. The `config.h` does not allow us to do this.
Example:
We have a define of `MUSESCORE_VERSION`, now it is global... and it is used directly everywhere, instead of using the `mu::String MUVersion::version()` method. I would like us to use only `mu::String MUVersion::version()` method everywhere, and `MUSESCORE_VERSION` define was only in the implementation of this method. that is, it needs to be made local to the global module, and made inaccessible to other modules. (now there are some technical problems related specifically to the global module, because of the pch headers, but another topic)

3. I want uniformity, so that there is limit approach to working with defines. 
There were three approaches:
* Technical defines, set in `SetupBuildEnvironment.cmake`
* Global logical defines defined in different cmake files
* Local logical defines in cmake of modules
* Defines in  `config.h`

Now:
* Technical defines, set in `SetupBuildEnvironment.cmake`
* Global logical defines set in `SetupConfigure.cmake` (should be)
* Local logical defines in cmake of modules